### PR TITLE
Add MessageSquare icon

### PR DESCRIPTION
**Pull request by joshxg**
Starts #158 

So I just wanted to PR this so we can see the problem:

image

In the zeplin:

image

- So, each row doesn't have a comment square and the popover isn't editable. Where do we edit these? At the moment the only place we could have this is in the `BatchesTable` - but I am not sure there is enough room?
- Then, since each line is an aggregate of batches, what comment shows?
- Could show it in the row expansion, whatever that is
- In mSupply, the note shows as a column in a CI (Not SI) but I can't find a way to edit it.. 🤷

### CODE REVIEW

**Comment by mark-prins** [comment]
I _think_ the idea is this

Screen Shot 2021-11-01 at 2 50 16 PM

[zeplin](https://app.zeplin.io/project/60e44fff6fa942168092f1ff/screen/610a7f901c36e2b128f4bb2c)

in which the comment is read only on hover and editable on click - at least for web. maybe just drop the hover for mobile? or use a long-press for edit?

File: packages/invoices/src/OutboundShipment/DetailView/reducer.ts [hunk 1]
```diff
@@ -152,6 +153,19 @@ export const reducer = (
           break;
         }
 
+        case ActionType.UpdateComment: {
+          const { payload } = action;
+          const { rowKey, comment } = payload;
```
Comment (mark-prins) [inline packages/invoices/src/OutboundShipment/DetailView/reducer.ts:158]: :nit I'd prefer `rowId` as this is matching against `id`

**Review by mark-prins — APPROVED** [review]
👍  I think it looks great!

**Comment by joshxg** [comment]
Sure

Still unsure what comment is to be shown (Maybe i'm still unclear on what the rows are) and also when the comment message icon shows. In the zepplin it appears to only be shown when there is a comment - I suppose we could be going for the mSupply style: a comment column and no way to edit it?

**Comment by mark-prins** [comment]
I think it should always show - maybe with the icon in orange if there is already a comment and in grey if there isn't one? Either way, clicking on it would bring up the edit modal.

Though...

The API doesn't have a comment of course - and in desktop there isn't one on the `trans_line` there's only the `item_line_note` - which would make sense perhaps, if we brought that through as a read only 'comment' for the stock line 🤷  In which case, the icon only shows if there is a note. this all fits in better with existing data

**Comment by joshxg** [comment]
Ah, is `[trans_line]note` the note from the `[item_line]`?

**Comment by joshxg** [comment]
Finally figured out how to show a note in mSupply! 🎉 

So the CI does show the `[item_line]note` 

since it seems each line represents multiple batches, I suppose we need to change the design to have subheadings for each batch?

**Comment by mark-prins** [comment]
aha. yes - I hadn't connected the dots.. the note is for an `[item_line]` not an `[item]`!
That would be a good option - have a heading of the batch (in bold) and then the note

**Pull request merged**